### PR TITLE
[DUOS-1282][risk=low] Post LC Fix

### PR DESCRIPTION
**Pull request by rfricke-asymmetrik**
**Addresses**
https://broadworkbench.atlassian.net/browse/DUOS-1282
**Changes**
- Change Library Card resource to look up library card user by email and ignore the userId field from the payload
- Update tests to accommodate

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing
- [ ] I've updated Swagger to reflect any API changes

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/resources/LibraryCardResource.java [hunk 1]
```diff
@@ -81,6 +83,13 @@ public Response createLibraryCard(@Auth AuthUser authUser, String libraryCard) {
     try{
       User user = userService.findUserByEmail(authUser.getName());
       LibraryCard payload = new Gson().fromJson(libraryCard, LibraryCard.class);
+      User lcUser = userService.findUserByEmail(payload.getUserEmail());
+
+      if (Objects.isNull(lcUser)) {
+        throw new BadRequestException("User email does not belong to a valid user.");
+      }
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/LibraryCardResource.java]: The service call will throw a NFE if the user is null, so this check should never happen.

**Review by rushtong — CHANGES_REQUESTED** [review]
Looks good. See comment inline about the exception case that will not be caught.

File: src/main/java/org/broadinstitute/consent/http/resources/LibraryCardResource.java (hunk 1 above)
Comment (rfricke-asymmetrik) [reply src/main/java/org/broadinstitute/consent/http/resources/LibraryCardResource.java]: Removed the unecessary thrown exception.

**Review by rushtong — APPROVED** [review]
Thank you for the changes 👍🏽

**Pull request merged**